apiserver/facadeversions_test.go

```diff
@@ -31,14 +31,72 @@ func (s *facadeVersionSuite) TestFacadeVersionsMatchServerVersions(c *gc.C) {
 	}
 	allServerFacades := apiserver.AllFacades().List()
 	serverFacadeNames := set.NewStrings()
-	serverFacadeBestVersions := make(map[string][]int, len(allServerFacades))
+	serverFacadeBestVersions := make(map[string]int, len(allServerFacades))
 	for _, facade := range allServerFacades {
 		serverFacadeNames.Add(facade.Name)
-		serverFacadeBestVersions[facade.Name] = facade.Versions
+		serverFacadeBestVersions[facade.Name] = facade.Versions[len(facade.Versions)-1]
```

Comment:
We need a comment here stating that the tested requirement is for the latest version to be known and supported.

We also need a comment either here or where the version declarations are to state that the order of supported versions must be low to high, otherwise this test becomes flaky.

Reply:
> We also need a comment either here or where the version declarations are to state that the order of supported versions must be low to high, otherwise this test becomes flaky.

Maybe we just sort them, which will remove the problem. We don't actually need them to be in order for it to pick the best version.